src/MessageScheduler.hpp

```diff
@@ -69,9 +71,10 @@ class MessageScheduler : public QObject {
 
   /**
    * @brief Fire this to indicate that the network is free
+   * Updates the counter for network backpressure
    */
-  void network_unblocked() {
-    is_network_unblocked = true;
+  void network_unblocked(uint64_t message_index, uint64_t last_ponged_index) {
+    network_backpressure_counter = std::min(message_index - last_ponged_index, max_queue_before_waiting);
```

Comment:
is it correct to compute the min with max_queue here? that means you'll always try to schedule on "network_unblocked" even if there is still a lot of backpressure